Feature/remove device

StefanEvanghelides opened the pull request:
Fixed #23 .
The DELETE request works properly.

Commit:
Merge pull request #8 from RUGSoftEng/merge_gui_communication

Release 0.2
Needs a proper OOP solution for encoding types,fragment class needs refactoring.

StefanEvanghelides committed:
Initial commit of the RemoveDevice feature

This will be used to fix #23 .
Right now it's a simple class trying to fire a DELETE request to the
server.
NOTE: unlike other features, this one does not extends AsyncTask. Might
come back later and implement it using AsyncTask.

StefanEvanghelides committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into feature/removeDevice

StefanEvanghelides committed:
Small fix on the RemoveDevice

Made the RemoveDeviceTask an AsyncTask.
Implemented a small bug because of which the App would crash when you
tried to remove a device. Now it doesn't crash, but the feature work
properly.

StefanEvanghelides committed:
Fixed #23 - DELETE request works properly

Removed the final "/" from the path. It fixed everything. Thanks Lars
for pointing it out.

FrankTN commented on Hestia/app/src/main/java/hestia/UI/ExpandableListAdapter.java:
```diff
@@ -105,7 +105,12 @@ public boolean onMenuItemClick(MenuItem item) {
                                 break;
                             case R.id.delete:
-                                //c.deleteDevice((Device)((DeviceBar) getChild(groupPosition, childPosition)).getDevice())
+                                /**
```
I would suggest removing this comment before merging into development. That way, we will keep our development branch clean.

StefanEvanghelides replied:
True, will add an issue to it. It would be nice to have a confirmation screen before actually deleting it.

FrankTN commented on Hestia/app/src/main/java/hestia/backend/RemoveDeviceTask.java:
```diff
@@ -0,0 +1,82 @@
+            if(httpCon != null) {
+                // FOR DEBUGGING ONLY -> PRINTING THE RESPONSE CODE
+
+
```
Create a new method for this (such as printResponseCode() ), to keep the main task simple and clean.

StefanEvanghelides replied:
yep. I should've taken another look at it. Looks horrible right now

FrankTN commented on Hestia/app/src/main/java/hestia/backend/RemoveDeviceTask.java:
```diff
@@ -0,0 +1,82 @@
+ * the Device "device".
+ */
+
+public class RemoveDeviceTask extends AsyncTask<Void, Void, Void> {
```
If you change the third type to Integer, you can have the task return the response code. This is preferable to just printing it and returning null.

StefanEvanghelides replied:
I don't agree with you here. there is no point of returning a value. you may argue that it would be neater this way, but if you return a value, you would then want to store it somewhere in CIC. It wouldn't be nice that way. You will be using this value only for Logging, but Logging should be done inside the RemoveDeviceTask class.

Also, think of it like this. you are sending a DELETE request. you shouldn't be returning anything, right?

FrankTN replied:
We are already returning the response code in the deviceRetrieverTask. This decision was made so that we can handle the response code at a later point. It might occur that the device can not be deleted. In this case, we want to be able to notify the user. Therefore, we decided to return the HTTP response code. I agree with you that currently it is not useful, but it might be at some point. 

If you don' t add it here then we should remove it everywhere. This should happen only after discussing it with everyone on the client team, so that we can ensure that everyone is on the same page. For now, I think we should add it. We can always discuss it at a meeting.

StefanEvanghelides replied:
I see. will add it then.

FrankTN commented on Hestia/app/src/main/java/hestia/backend/RemoveDeviceTask.java:
```diff
@@ -0,0 +1,82 @@
+                httpCon.disconnect();
+            }
+        }
+        return null;
```
Return the response code

FrankTN commented on Hestia/app/src/main/java/hestia/backend/RemoveDeviceTask.java:
```diff
@@ -0,0 +1,82 @@
+        return null;
+    }
+
+    /**
```
I would not store the path to the device like this. This way, it is not clear that the path is not general but points to the specific device. I suggest removing this function. I don' t think you need to store this path as its only used once.

StefanEvanghelides replied:
you are right. will change this asap.

FrankTN reviewed (changes requested):
Looks good overall. I have commented on some minor issues which need fixing before this can be merged into development.

StefanEvanghelides committed:
Refactoring

Refactored the code

StefanEvanghelides committed:
removed comment

Removed the comment about using a pop-up window to confirm the deletion.
Will create an issue for it

StefanEvanghelides committed:
Deleted the logResponse() function

Deleted the logResponse() function. This action will be done
externatlly, if needed

FrankTN reviewed (approved):
Looks good, this can be merged with development.
